pkg/scheduler/framework/interface.go

```diff
@@ -159,6 +159,15 @@ type Status struct {
 	failedPlugin string
 }
 
+func (s *Status) WithError(err error) *Status {
+	if err == nil {
+		return s
```

Comment:
This works like a build pattern, like `status = NewStatus().WithError(err).WithSomethingElse(xxx)`. When we want to construct a unschedulable status with fitError, this can help. We already have one `AsStatus()` which will return an error wrapped new status.

Reply:
On second thought.... let's remove lines 163 to 165.

We don't want people to accidentally use this function like this:

```golang
var s *Status
s.WithError(err)
```

I would rather get a segmentation fault.